pkg/messaging/broker/pubsub.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package broker
+
+import (
+	"github.com/mainflux/mainflux"
+	log "github.com/mainflux/mainflux/logger"
+
+	"github.com/mainflux/mainflux/pkg/messaging/nats"
+	"github.com/mainflux/mainflux/pkg/messaging/rabbitmq"
+)
+
+const (
+	// chansPrefix = "channels"
+
+	// SubjectAllChannels represents subject to subscribe for all the channels.
+	SubjectAllChannels = "channels.>"
+)
+
+// PubSub type
+type PubSub nats.PubSub
+
+// NewPubSub This aggregates the NewPubSub function for all message brokers
+func NewPubSub(url, queue string, logger log.Logger) (nats.PubSub, error) {
```

Comment:
You can join these files (`publisher.go` and `pubsub.go`) to a single `pkg/messaging/brokers/brokers.go`.

Reply:
Sorry this is in the Kafka branch